### PR TITLE
Sawa 164 make achievement class

### PR DESCRIPTION
**Pull request by fabiogiorgio**
This is what I got for the implementation of the achievements, no logic yet.

Keep in mind the following:

- I added a repository layer
- I added a service layer

Good or bad idea? Please let me know

### CODE REVIEW

File: ShowerShow/Repository/Interfaces/IAchievementRepository.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using ShowerShow.Models;
```
Comment (egecnr) [inline ShowerShow/Repository/Interfaces/IAchievementRepository.cs:1]: Ideally the folder names should not be plural. Changing this to interface might be better.

File: ShowerShow/Repository/Interfaces/IAchievementRepository.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ShowerShow.Repository.Interfaces
+{
+    public interface IAchievementRepository
+    {
+        Task<List<Achievement>> getAchievementsById(Guid userId); //getting a list of achievement
```
Comment (egecnr) [inline ShowerShow/Repository/Interfaces/IAchievementRepository.cs]: The Method names in repo and service shouldnt be camel case:  getAchivementsById -> GetAchivementsById. If you can check this in other places you wrote as well thatd be great. Good job with the architecture btw I'll also implement it similar to this. (Get hamburgered son lol)

File: ShowerShow/Repository/Interfaces/IAchievementService.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ShowerShow.Repository.Interfaces
+{
+    internal interface IAchievementService
+    {
+        Task<List<Achievement>> getAchievementsById(Guid userId); //getting a list of achievement
+        Task<Achievement> getAchievementById(Guid achievementId); //getting a single achievement
```
Comment (egecnr) [inline ShowerShow/Repository/Interfaces/IAchievementService.cs]: Same story here. you got it

File: ShowerShow/Service/AchievementService.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+{
+    internal class AchievementService : IAchievementService
+    {
+        private readonly IAchievementRepository _achievementRepository;
+
+        public AchievementService(IAchievementRepository achievementrepository)
+        {
+            _achievementRepository = achievementrepository;
+        }
+        public Task<Achievement> getAchievementById(Guid achievementId)
```
Comment (egecnr) [inline ShowerShow/Service/AchievementService.cs]: Same here

File: ShowerShow/Model/Achievement.cs [hunk 5]
```diff
@@ -17,5 +16,17 @@ public class Achievement
         public int CurrentValue { get; set; } = 0;
         [JsonRequired]
         public int TargetValue { get; set; }
+
+
+        public Achievement(string title, string description, int currentValue, int targetValue)
+        {
+
+            //this.Id = id;
```
Comment (egecnr) [inline ShowerShow/Model/Achievement.cs]: You can delete this comment

**Review by egecnr — COMMENTED** [review]
GGWP

**Comment by logoffro1** [comment]
Besides what Ege said, there might be a few changes in the architecture but we don't know yet how that would look, so for now you did a great job!

**Comment by logoffro1** [comment]
Can you make the PR again and put it into DEV instead of MAIN ?

File: ShowerShow/Repository/Interfaces/IAchievementRepository.cs (hunk 1 above)
Comment (fabiogiorgio) [reply ShowerShow/Repository/Interfaces/IAchievementRepository.cs:1]: image

Comment (fabiogiorgio) [reply ShowerShow/Repository/Interfaces/IAchievementRepository.cs:1]: image

Comment (fabiogiorgio) [reply ShowerShow/Repository/Interfaces/IAchievementRepository.cs:1]: image

Comment (fabiogiorgio) [reply ShowerShow/Repository/Interfaces/IAchievementRepository.cs:1]: image

File: ShowerShow/Control/AchievementController.cs [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        {
+            _logger = logger;
+            _achievementService = service;
+        }
+
+        [Function(nameof(GetAchievementsById))] 
+        [OpenApiOperation(operationId: "GetUserAchievements", tags: new[] { "Achievement" })]
+        [OpenApiParameter(name: "UserId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The User ID parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(List<Achievement>), Description = "The OK response with all achievements from user.")]
+        public async Task<IActionResult> GetAchievementsById([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "user/{UserId}/achievements")] HttpRequestData req, Guid UserId)
```
Comment (egecnr) [inline ShowerShow/Control/AchievementController.cs]: IactionResult has to change to HttpResponseData as a return type for all endpoints now. Its a recent development

File: ShowerShow/Repository/Interfaces/IAchievementRepository.cs (hunk 1 above)
Comment (egecnr) [reply ShowerShow/Repository/Interfaces/IAchievementRepository.cs:1]: Those we need plural i think lol :D. Not too sure i'll look into it

**Comment by egecnr** [comment]
You can delete this PR and make a new one once you are done with endpoints @fabiogiorgio

**Pull request closed without merging**